header/p2p/server.go

```diff
@@ -124,6 +137,10 @@ func (serv *ExchangeServer) handleRequest(from, to uint64, stream network.Stream
 		}
 		headers = headersByRange
 	}
+
+	if err := stream.SetWriteDeadline(time.Now().Add(writeDeadline * time.Duration(len(headers)))); err != nil {
```

Comment:
Does the write deadline persist per-write? If not, we should just set it before writing, but we shouldn't make the write deadline so long.
```suggestion
	if err := stream.SetWriteDeadline(time.Now().Add(writeDeadline)); err != nil {
```

Reply:
No, it can't be set per-write. But the case that you suggested assumes that timeouts for writing 1 block or 200 blocks are equal. Is it correct?

Reply:
The thing is, you are performing 1 write per header that is being written bc it happens in a loop. The write deadline shouldn't be too long on that. So if you set the write deadline on the given stream, does it persist on the stream for that host or does it get "cleared" after one write has been made?

Reply:
It not cleared. To clear it and empty time should be passed to it. 

@vgonkivs in fraudsync we didn't care about resetting the deadline because we closed it straight after the one only write. If we reuse the stream for multiple writes, then deadlines must be reset for both reads and writes

Reply:
ok